portfolio/src/main/webapp/script.js

```diff
@@ -11,40 +11,81 @@
-
-function getMessagesFromServer() {
-  var numberOfComments = document.getElementById("messages-number").value;
-  if(numberOfComments === null){
+async function getCommentsFromServer() {
+  var numberOfComments = document.getElementById("comments-number").value;
+  if(numberOfComments == null){
       numberOfComments = 0;
   }
   
   var url = '/data?nr=' + numberOfComments;
-  console.log(numberOfComments);
-  fetch(url).then(response => response.json()).then((messages) => {
-    console.log(messages);
-    const messagesListElement = document.getElementById('messages-container');
+  return fetch(url).then(response =>response.json());
+  
+}
+
+
+async function handleGetCommentsClick(){
+   const messages = await getCommentsFromServer();
+   renderComments(messages);
+}
+
+async function renderComments(messages){
+    console.log("2:" + messages);
+    const messagesListElement = document.getElementById('comments-container');
     if(messagesListElement.innerHTML !== ''){
         messagesListElement.innerHTML = '';
     }
-    for(i = 0; i < messages.length; ++i){
-        messagesListElement.appendChild(createListElement(messages[i]));
+
+    for(var i = 0; i < messages.length; ++i){
+        var domListElement = createDOMListElement(messages[i].propertyMap.email, messages[i].propertyMap.content);
+        domListElement.appendChild(createDOMButton(messages[i].propertyMap.messageId));
+        messagesListElement.appendChild(domListElement);
+    }
+    
+    if(messages.length === 0){
+        var content = "No comments available!";
+        var domListElement = createDOMListElement(content);
+        messagesListElement.appendChild(domListElement);
     }
-  });
 }
 
-function createListElement(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  liElement.classList.add('list-group-item');
-  return liElement;
+function createDOMListElement(email, text) {
+  const domListElement = document.createElement('li');
+  if(email != "")
+   domListElement.innerText = email + ": ";
+  domListElement.innerText += text;
+  domListElement.classList.add('list-group-item');
+  return domListElement;
 }
 
+function createDOMButton(messageId) {
+  const domButtonElement = document.createElement('button');
+  domButtonElement.classList.add('delete-button');
+  domButtonElement.textContent = "Delete";
+  domButtonElement.id = messageId;
+  domButtonElement.setAttribute( "onClick", "javascript: handleDeleteCommentClick(this);" );
+  return domButtonElement;
+}
+
+async function handleDeleteCommentClick(thisButton){
+   deleteComment(thisButton);
+   const messages = await getCommentsFromServer();
+   renderComments(messages);
+}
+
+function deleteComment(thisButton) {
+  const params = new URLSearchParams();
+  var messageId = thisButton.id;
+  var liElementToDelete = thisButton.parentNode;
+  params.append('messageId', messageId);
+  fetch('/delete-data', {method: 'POST', body: params});
```

Comment:
You don't need to delete the comment from the DOM if you're doing renderComments(messages) afterwards.

Also, check for the fetch response, you should only continue if the deletion on the backend was successful, otherwise show the error (if possible).

Reply:
Did you act on this? I don't see new changes.

Reply:
I don't have to literally delete the <li>, I re-render the comments. So, I've removed that part.